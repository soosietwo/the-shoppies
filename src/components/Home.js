import React, { useContext, useCallback, useState } from "react";
import { Page, Card, Frame, Banner } from "@shopify/polaris";
import { gql, useQuery } from "@apollo/client";

import MoviesList from "./MoviesList";
import Pagination from "./Pagination";
import Loader from "./Loader";
import Error from "./Error";
import Sidebar from "./Sidebar";
import Header from "./Header";
import MovieCard from "./MovieCard";
import { Context } from "../store";

export const NOMINEES_QUERY = gql`
  query NOMINEES_QUERY {
    nominees {
      id
      title
      poster
      year
    }
  }
`;

const Home = () => {
  const [state] = useContext(Context);
  const [sheetActive, setSheetActive] = useState(false);
  const toggleSheetActive = useCallback(
    () => setSheetActive((sheetActive) => !sheetActive),
    []
  );

  const { loading, error, data } = useQuery(NOMINEES_QUERY);

  return (
    <Frame topBar={<Header toggleSheetActive={toggleSheetActive} />}>
      <Page>
        <Card>
          {state.error && !state.isLoading && (
            <Error title="Oops! Something went wrong" details={state.error} />
          )}

          {data && data.nominees.length === 5 && (
            <Banner
              status="success"
              title="All done!"
              action={{
                content: "Go to nominees",
                onAction: () => toggleSheetActive(true),
              }}
            >
              <p>You have chosen 5 nominees. Submit or remove them.</p>
            </Banner>
          )}

          {state.isLoading || loading ? (
            <Loader count={10} />
          ) : (
            <MoviesList
              singular="movie"
              plural="movies"
              movies={state.movies}
              totalResults={state.totalResults}
              emptyStateTitle="No movies found"
              emptyStateCaption="Start typing in your favourite movie titles to get started!"
              renderItem={(movie) => (
                <MovieCard
                  movie={movie}
                  nominees={(data && data.nominees) || []}
                  toggleSheetActive={toggleSheetActive}
                />
              )}
            />
          )}

          {state.totalResults > 10 && <Pagination />}
        </Card>

        <Sidebar
          nominees={(data && data.nominees) || []}
          toggleSheetActive={toggleSheetActive}
          sheetActive={sheetActive}
        />
      </Page>
    </Frame>
  );
};

export default Home;
