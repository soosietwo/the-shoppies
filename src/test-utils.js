// test-utils.js
import React from "react";
import { render } from "@testing-library/react";
import { AppProvider, Frame } from "@shopify/polaris";
import en from "@shopify/polaris/locales/en.json";
import Store from "./store";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const AllTheProviders = ({ children }) => {
  return (
    <AppProvider i18n={en}>
      <Store>
        <Frame>{children}</Frame>
      </Store>
    </AppProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
