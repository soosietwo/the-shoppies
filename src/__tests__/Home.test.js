import React from "react";
import { render } from "../test-utils";
import { MockedProvider } from "@apollo/client/testing";
import Store from "../store";
import Home from "../components/Home";

describe("Home", () => {
  it("renders and matches snapshot", () => {
    const { container } = render(
      <MockedProvider>
        <Store>
          <Home />
        </Store>
      </MockedProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
