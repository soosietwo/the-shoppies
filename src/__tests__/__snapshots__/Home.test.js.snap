// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders and matches snapshot 1`] = `
<div>
  <div>
    <div
      class="Polaris-Frame"
      data-polaris-layer="true"
    >
      <div
        class="Polaris-Frame__Skip"
      >
        <a
          class="Polaris-Button"
          data-polaris-unstyled="true"
          data-polaris-unstyled-button="true"
          href="#AppFrameMainContent"
        >
          <span
            class="Polaris-Button__Content"
          >
            <span
              class="Polaris-Button__Text"
            >
              Skip to content
            </span>
          </span>
        </a>
      </div>
      <div
        class="Polaris-Frame__ContextualSaveBar Polaris-Frame-CSSAnimation--startFade"
      />
      <main
        class="Polaris-Frame__Main"
        data-has-global-ribbon="false"
        id="AppFrameMain"
      >
        <a
          id="AppFrameMainContent"
          tabindex="-1"
        />
        <div
          class="Polaris-Frame__Content"
        >
          <div
            class="Polaris-Frame Polaris-Frame--hasTopBar"
            data-polaris-layer="true"
          >
            <div
              class="Polaris-Frame__Skip"
            >
              <a
                class="Polaris-Button"
                data-polaris-unstyled="true"
                data-polaris-unstyled-button="true"
                href="#AppFrameMainContent"
              >
                <span
                  class="Polaris-Button__Content"
                >
                  <span
                    class="Polaris-Button__Text"
                  >
                    Skip to content
                  </span>
                </span>
              </a>
            </div>
            <div
              class="Polaris-Frame__TopBar"
              data-polaris-layer="true"
              data-polaris-top-bar="true"
              id="AppFrameTopBar"
            >
              <div
                class="Polaris-TopBar"
              >
                <button
                  aria-label="Toggle menu"
                  class="Polaris-TopBar__NavigationIcon"
                  type="button"
                >
                  <span
                    class="Polaris-Icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="Polaris-Icon__Svg"
                      focusable="false"
                      viewBox="0 0 20 20"
                    >
                      <path
                        d="M19 11H1a1 1 0 0 1 0-2h18a1 1 0 1 1 0 2zm0-7H1a1 1 0 0 1 0-2h18a1 1 0 1 1 0 2zm0 14H1a1 1 0 0 1 0-2h18a1 1 0 0 1 0 2z"
                      />
                    </svg>
                  </span>
                </button>
                <div
                  class="Polaris-TopBar__Contents"
                >
                  <div
                    class="Polaris-TopBar__SearchField"
                  >
                    <div
                      class="Polaris-TopBar-SearchField"
                    >
                      <span
                        class="Polaris-VisuallyHidden"
                      >
                        <label
                          for="PolarisSearchField1"
                        >
                          Search
                        </label>
                      </span>
                      <input
                        autocapitalize="off"
                        autocomplete="off"
                        autocorrect="off"
                        class="Polaris-TopBar-SearchField__Input"
                        id="PolarisSearchField1"
                        placeholder="Search"
                        type="search"
                        value=""
                      />
                      <span
                        class="Polaris-TopBar-SearchField__Icon"
                      >
                        <span
                          class="Polaris-Icon"
                        >
                          <svg
                            aria-hidden="true"
                            class="Polaris-Icon__Svg"
                            focusable="false"
                            viewBox="0 0 20 20"
                          >
                            <path
                              d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8zm9.707 4.293l-4.82-4.82A5.968 5.968 0 0 0 14 8 6 6 0 0 0 2 8a6 6 0 0 0 6 6 5.968 5.968 0 0 0 3.473-1.113l4.82 4.82a.997.997 0 0 0 1.414 0 .999.999 0 0 0 0-1.414z"
                            />
                          </svg>
                        </span>
                      </span>
                      <div
                        class="Polaris-TopBar-SearchField__Backdrop Polaris-TopBar-SearchField__BackdropShowFocusBorder"
                      />
                    </div>
                  </div>
                  <div
                    class="Polaris-TopBar__SecondaryMenu"
                  >
                    <div>
                      <div
                        class="Polaris-TopBar-Menu__ActivatorWrapper"
                      >
                        <button
                          aria-controls="Polarispopover1"
                          aria-expanded="false"
                          aria-owns="Polarispopover1"
                          class="Polaris-TopBar-Menu__Activator"
                          tabindex="0"
                          type="button"
                        >
                          <div
                            style="display: flex; align-items: center;"
                          >
                            Nominees
                            <span
                              class="sc-gtssRu hdlOKH"
                            >
                              <div>
                                <div
                                  class="sc-bdnylx czgcBA count"
                                >
                                  0
                                </div>
                              </div>
                            </span>
                          </div>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="Polaris-Frame__ContextualSaveBar Polaris-Frame-CSSAnimation--startFade"
            />
            <main
              class="Polaris-Frame__Main"
              data-has-global-ribbon="false"
              id="AppFrameMain"
            >
              <a
                id="AppFrameMainContent"
                tabindex="-1"
              />
              <div
                class="Polaris-Frame__Content"
              >
                <div
                  class="Polaris-Page"
                >
                  <div
                    class="Polaris-Page__Content"
                  >
                    <div
                      class="Polaris-Card"
                    >
                      <div
                        class="Polaris-ResourceList__ResourceListWrapper"
                      >
                        <div
                          class="Polaris-Card"
                        >
                          <div
                            class="Polaris-EmptyState Polaris-EmptyState--withinContentContainer"
                          >
                            <div
                              class="Polaris-EmptyState__Section"
                            >
                              <div
                                class="Polaris-EmptyState__DetailsContainer"
                              >
                                <div
                                  class="Polaris-EmptyState__Details"
                                >
                                  <div
                                    class="Polaris-TextContainer"
                                  >
                                    <p
                                      class="Polaris-DisplayText Polaris-DisplayText--sizeSmall"
                                    >
                                      No movies found
                                    </p>
                                    <div
                                      class="Polaris-EmptyState__Content"
                                    >
                                      <p>
                                        Start typing in your favourite movie titles to get started!
                                      </p>
                                    </div>
                                  </div>
                                </div>
                              </div>
                              <div
                                class="Polaris-EmptyState__ImageContainer"
                              >
                                <img
                                  alt=""
                                  class="Polaris-EmptyState__Image"
                                  role="presentation"
                                  src="https://www.iconarchive.com/download/i99782/designbolts/free-multimedia/Film.ico"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </main>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
